### PR TITLE
1st Draft

### PR DESCRIPTION
**Pull request by seanturner**

### CODE REVIEW

**Comment by seanturner** [comment]
I should note that this is supposed to be a light touch set of changes.

File: wg-charter/01-01 [hunk 1]
```diff
@@ -39,35 +31,29 @@ different applications can adapt to their own needs.
...
+at RFC, we had several interoperable implementations as well as a thorough
+security analysis.
 
 Note that consensus is required both for changes to the protocol mechanisms
 from these documents and retention of the mechanisms from them. In particular,
 because something is in the initial document set does not imply that there is
 consensus around the feature or around how it is specified.
+
+Now that MLS has been published the group will work on MLS protocol extensions
+and cipher suites.
```
Comment (bifurcation) [inline wg-charter/01-01]: Overall, I think these changes are good.  I agree with @ekr's comments at IETF 118 that we should have some scoping around what extensions we will develop.  I surveyed the existing proposals, and I think they point to a few buckets:

```suggestion
Now that MLS has been published the group will work on the following MLS protocol extensions:

* Support for use of MLS in protocols developed by the MIMI working group
* Support for Verifiable Credentials
* Support for common operational patterns in messaging applications
* Quantum-safe cipher suites
* Framework for safe extensibility
* Detection of lost application messages
* Support for sending messages to individual members of a group
```

This taxonomy is based on the current document set:

* Support for use of MLS in protocols developed by the MIMI working group
    * SelfRemove proposal (draft-mahy-mlm-selfremove)
    * Room Policy (draft-mahy-mls-room-policy-ext)
    * Content Advertisement (draft-ietf-mls-extensions)
* Support for Verifiable Credentials
    * UserInfo VC (draft-barnes-mls-addl-creds)
    * Multi-Cred (draft-barnes-mls-addl-creds)
    * Weak Multi-Cred (draft-barnes-mls-addl-creds)
* Support for common operational patterns in messaging applications
    * KeyPackage context (draft-mahy-mls-kp-context)
    * Last resort KeyPackages (draft-ietf-mls-extensions)
* Quantum-safe ciphersuites
    * PQ-Hybrid KEM (draft-mahy-mls-x25519kyber768draft00)
* Framework for safe extensibility
    * Safe Extensions (draft-ietf-mls-extensions)
* Detection of lost application messages
    * AppAck (draft-ietf-mls-extensions)
* Support for sending messages to individual members of a group
    * Targeted Messages (draft-ietf-mls-extensions)

**Review by bifurcation — APPROVED** [review]

**Comment by seanturner** [comment]
@bifurcation thanks for kicking off the edits with this list. Should we also add something like that I think addresses ekr's other point:

    Work on other items requires a recharter.

**Comment by bifurcation** [comment]
That limitation SGTM

**Comment by seanturner** [comment]
And, I guess we'll need to put dates on 'em too.

**Comment by br-hale** [comment]
We are expecting drafts on strengthening PCS when an MLS member has multiple devices (guardianship - as presented at IETF117 and mentioned in the works at IETF118) as well as hybrid quantum resistance. The latter does not fall under a 'Quantum-safe ciphersuites' descriptor though, as achieving that may entail more than a ciphersuite descriptor.

File: wg-charter/01-01 (hunk 1 above)
Comment (rohan-wire) [reply wg-charter/01-01]: I think this is a good list, but I feel that SelfRemove and Content Advertisement is part of "a common operational pattern in messaging". (That does not change the proposed text however)

**Comment by rohan-wire** [comment]
Brita said:
> We are expecting drafts on strengthening PCS when an MLS member has multiple devices (guardianship - as presented at IETF117 and mentioned in the works at IETF118) as well as hybrid quantum resistance. The latter does not fall under a 'Quantum-safe ciphersuites' descriptor though, as achieving that may entail more than a ciphersuite descriptor.

I think we could create a category of Post-quantum support, with multiple milestones, one of which is "PQ ciphersuite addressing immediate need for harvest-now, decrypt-later case"

Sean said:
> And, I guess we'll need to put dates on 'em too.

For the MIMI items, we have to be in sync with them. We can use their latest date (March 2024) from their charter for this item, but I expect these will be much later.

How about July 2024 for common operational patterns, safe extensibility, and the first PQ ciphersuite?

How about Nov 2024 for AppAck, Targeted messages, and Credentials? I think we may revise AppAck based on a  requirements and scalability discussion, targeted messages is complicated enough that it may require the WG more time to understand the proposal, the VC credentials have a dependency on a non-IETF document which should be stable/ready as we request publication.

**Comment by br-hale** [comment]
> I think we could create a category of Post-quantum support, with multiple milestones, one of which is "PQ ciphersuite addressing immediate need for harvest-now, decrypt-later case"

A general 'PQ support" direction would be useful with sub-goals:
- 'generic PQ support' (i.e. with a PQ-only ciphersuite - we do not have this, do we want it for MLS?)
- 'hybrid PQ support' (current https://www.ietf.org/id/draft-mahy-mls-x25519kyber768draft00-00.txt)
- 'efficiency-preserving hybrid PQ support' (coming draft)

I recommend against claiming just one of these is for addressing the "harvest-now, decrypt-later case", and rather keeping simple descriptions on differences.

**Comment by ekr** [comment]
I don't think we should do a PQ-only ciphersuite. In the absence of the draft it's hard to know what to think of "efficiency-preserving hybrid PQ"

**Comment by seanturner** [comment]
Since the bullets do not track to an individual I-D, I prefer the idea of leaving it at a higher lever "Post-quantum support".

As for milestones, I will take a stab at this but since we are not sure what extensions will end up in which I-Ds I kind of prefer putting some disclaimers in about some appearing in mls-extensions and others appearing elsewhere. Anyway, I'll take a stab at this.

**Comment by seanturner** [comment]
Whoops committed directly and shouldn't have. Changes are here:
https://github.com/mlswg/wg-materials/pull/14/commits/d7cab2f6ffe5c59d18cea7b81ddbdf82fb6c4c65

File: wg-charter/01-01 [hunk 2]
```diff
@@ -39,35 +31,47 @@ different applications can adapt to their own needs.
...
 Note that consensus is required both for changes to the protocol mechanisms
 from these documents and retention of the mechanisms from them. In particular,
 because something is in the initial document set does not imply that there is
 consensus around the feature or around how it is specified.
+
+Now that MLS has been published the group will work on the following MLS
+protocol extensions:
+
+* Support for use of MLS in protocols developed by the MIMI working group
+* Support for Verifiable Credentials
```
Comment (seanturner) [inline wg-charter/01-01]:
```suggestion
* Support for new credential types
```

File: wg-charter/01-01 [hunk 3]
```diff
@@ -39,35 +31,47 @@ different applications can adapt to their own needs.
...
+implementation, and verification proceeding in parallel.  When we arrived
+at RFC, we had several interoperable implementations as well as a thorough
+security analysis.
 
 Note that consensus is required both for changes to the protocol mechanisms
 from these documents and retention of the mechanisms from them. In particular,
 because something is in the initial document set does not imply that there is
 consensus around the feature or around how it is specified.
+
+Now that MLS has been published the group will work on the following MLS
```
Comment (bifurcation) [inline wg-charter/01-01]:
```suggestion
Now that MLS has been published, the group will work on the following MLS
```

File: wg-charter/01-01 [hunk 4]
```diff
@@ -39,35 +31,47 @@ different applications can adapt to their own needs.
...
+protocol extensions:
+
+* Support for use of MLS in protocols developed by the MIMI working group
+* Support for new credential types
+* Support for common operational patterns in messaging applications
+* Support for Post-quantum
```
Comment (bifurcation) [inline wg-charter/01-01]:
```suggestion
* Support for quantum-safe cryptographic algorithms
```

File: wg-charter/01-01 [hunk 5]
```diff
@@ -39,35 +31,47 @@ different applications can adapt to their own needs.
...
+* Framework for safe extensibility
+* Detection of lost application messages
+* Support for sending messages to individual members of a group
+
+Many of extensions to support these features will be included in
+draft-ietf-mls-extensions, but some of the extensions will be published in
+seperate Internet-Drafts.
+
+Work on other items requires a recharter.
```
Comment (bifurcation) [inline wg-charter/01-01]:
```suggestion
```

This is true regardless of what the charter says.

**Review by bifurcation — APPROVED** [review]
Couple of editorial suggestions, otherwise lgtm.

File: wg-charter/01-01 (hunk 4 above)
Comment (br-hale) [reply wg-charter/01-01]: Quantum resistance work is not simply about the algorithms, so we should not limit it to that. The suggested change would also notably undo the edits discussed in the PR issue.

Comment (br-hale) [reply wg-charter/01-01]: If it is the word 'post-quantum' that the suggestion is about, then a change to "Support for quantum resistance" should suffice.

File: wg-charter/01-01 [hunk 6]
```diff
@@ -39,35 +31,47 @@ different applications can adapt to their own needs.
...
+protocol extensions:
+
+* Support for use of MLS in protocols developed by the MIMI working group
+* Support for new credential types
+* Support for common operational patterns in messaging applications
+* Support for Post-quantum
+* Framework for safe extensibility
+* Detection of lost application messages
+* Support for sending messages to individual members of a group
+
```
Comment (beurdouche) [inline wg-charter/01-01:68]:
```suggestion
* Support for end-to-end encrypted long term storage relying on group secrets
```

**Review by beurdouche — APPROVED** [review]
Added a suggestion, but overall looks good to me.
Should we mention extensions needed by MIMI or is that too loose ?

File: wg-charter/01-01 (hunk 4 above)
Comment (seanturner) [reply wg-charter/01-01]: I like this as it also addresses ekr's comment.

File: wg-charter/01-01 [hunk 7]
```diff
@@ -39,35 +31,46 @@ different applications can adapt to their own needs.
...
 because something is in the initial document set does not imply that there is
 consensus around the feature or around how it is specified.
+
+Now that MLS has been published, the group will work on the following MLS
+protocol extensions:
+
+* Support for use of MLS in protocols developed by the MIMI working group
+* Support for new credential types
+* Support for common operational patterns in messaging applications
+* Support for Post-quantum
```
Comment (seanturner) [inline wg-charter/01-01]:
```suggestion
* Support for quantum resistance
```

**Pull request merged**